recipe-box-backend/src/main/java/com/endava/recipebox/controller/RecipeController.java

```diff
@@ -35,5 +38,14 @@ public ResponseEntity<List<RecipeDTO>> getAllPublicRecipesByName(@RequestParam S
         return ResponseEntity.ok(recipeService.getAllPublicRecipesByName(recipeName));
     }
 
+    @PostMapping("/create")
+    public ResponseEntity<Recipe> createRecipe(@RequestBody RecipeRequestDTO recipeRequestDTO, @RequestParam Long userId) {
+        if (recipeRequestDTO != null && userId != null) {
```

Comment:
You can use apart from @RequestBody the annotation @Valid. And inside the DTO class to add validations like:
@NotNull
    @NotBlank
    private String name;
    @NotNull
    @NotBlank
    private String description;
    @NotNull
    @NotBlank
    @Pattern(regexp = "breakfast|lunch|dinner|dessert|snack", flags = Pattern.Flag.CASE_INSENSITIVE)
    private String type;
    @NotNull
    @NotBlank
    private String cookingTime;
    @NotNull
    @NotBlank
    @Pattern(regexp = "easy|medium|advanced", flags = Pattern.Flag.CASE_INSENSITIVE)
    private String difficulty;
    @NotNull
    @NotBlank
    private String steps;
    @NotNull
    @Range(min=1, max=4)
    private Integer servings;